Resources/config/security.xml

```diff
@@ -45,6 +45,8 @@
         </service>
 
         <service id="FOS\UserBundle\Controller\SecurityController" alias="fos_user.security.controller" public="true" />
+
+        <service id="fos_user.user_checker" class="FOS\UserBundle\Security\UserChecker" public="true" />
```

Comment:
no need to make it public, as it is meant to be referenced in the SecurityBundle config, which supports private services

Reply:
Moreover I would use FQCN as `id` in order to avoid aliasing this for bundle users that would like to take advantage of autowire. (i.e.: create a decorator for this user checker as I've suggested above)